src/server/qgsmapserviceexception.h

```diff
@@ -34,15 +36,12 @@
  *  * "OperationNotSupported"
  */
 
-class SERVER_EXPORT QgsMapServiceException : public QgsException
+class SERVER_EXPORT QgsMapServiceException : public QgsOgcServiceException
 {
   public:
-    QgsMapServiceException( const QString& code, const QString& message );
-    QString code() const {return mCode;}
-    QString message() const {return mMessage;}
-  private:
-    QString mCode;
-    QString mMessage;
+    QgsMapServiceException( const QString& code, const QString& message )
```

Comment:
This is the idea  at the end, but this is still heavily used in wfs, wcs:  I plan to remove QgsMapServiceException when all services will be moved as new services modules.